test/integration/good/code-gen/closures/cpp.expected

```diff
@@ -0,0 +1,4995 @@
+
+using stan::io::dump;
+using stan::model::assign;
+using stan::model::index_uni;
+using stan::model::index_max;
+using stan::model::index_min;
+using stan::model::index_min_max;
+using stan::model::index_multi;
+using stan::model::index_omni;
+using stan::model::model_base_crtp;
+using stan::model::rvalue;
+using namespace stan::math;
+
+
+stan::math::profile_map profiles__;
+static constexpr std::array<const char*, 30> locations_array__ = 
+{" (found before start of program)",
+ " (in 'capture_types.stan', line 8, column 4 to line 14, column 5)",
+ " (in 'capture_types.stan', line 17, column 4 to column 11)",
+ " (in 'capture_types.stan', line 18, column 4 to column 15)",
+ " (in 'capture_types.stan', line 19, column 4 to column 17)",
+ " (in 'capture_types.stan', line 22, column 4 to line 30, column 5)",
+ " (in 'capture_types.stan', line 31, column 4 to column 38)",
+ " (in 'capture_types.stan', line 32, column 4 to column 36)",
+ " (in 'capture_types.stan', line 33, column 4 to column 49)",
+ " (in 'capture_types.stan', line 34, column 4 to column 47)",
+ " (in 'capture_types.stan', line 2, column 4 to column 10)",
+ " (in 'capture_types.stan', line 3, column 4 to column 11)",
+ " (in 'capture_types.stan', line 4, column 11 to column 12)",
+ " (in 'capture_types.stan', line 4, column 4 to column 14)",
+ " (in 'capture_types.stan', line 5, column 11 to column 12)",
+ " (in 'capture_types.stan', line 5, column 4 to column 16)",
+ " (in 'capture_types.stan', line 18, column 12 to column 13)",
+ " (in 'capture_types.stan', line 19, column 11 to column 12)",
+ " (in 'capture_types.stan', line 10, column 8 to column 25)",
+ " (in 'capture_types.stan', line 11, column 8 to column 25)",
+ " (in 'capture_types.stan', line 12, column 8 to column 25)",
+ " (in 'capture_types.stan', line 13, column 8 to column 36)",
+ " (in 'capture_types.stan', line 9, column 44 to line 14, column 5)",
+ " (in 'capture_types.stan', line 24, column 8 to column 25)",
+ " (in 'capture_types.stan', line 25, column 8 to column 25)",
+ " (in 'capture_types.stan', line 26, column 8 to column 25)",
+ " (in 'capture_types.stan', line 27, column 8 to column 27)",
+ " (in 'capture_types.stan', line 28, column 8 to column 27)",
+ " (in 'capture_types.stan', line 29, column 8 to column 52)",
+ " (in 'capture_types.stan', line 23, column 55 to line 30, column 5)"};
+
+template <typename T0__, typename T3__, typename T4__>
+stan::promote_args_t<stan::value_type_t<T0__>, T3__,
+stan::value_type_t<T4__>>
+capture_data_L9C9_impl__(const T0__& v, const double& x,
+                         const std::vector<double>& y, const T3__& z,
+                         const T4__& r, std::ostream* pstream__)  ;
+
+class capture_data_L9C9_cfunctor__ {
+  const Eigen::Matrix<double, -1, 1>& v;
+  double x;
+  const std::vector<double>& y;
+  public:
+  const size_t vars_count__;
+  capture_data_L9C9_cfunctor__(const capture_data_L9C9_cfunctor__&) = default ;
```

Comment:
Does the compiler not handle these automatically?

Reply:
Probably? I think I was just playing it safe here. Not sure if the constructor is even needed anywhere...

Reply:
word yeah idt these are necessary. If you don't define any of the default generated constructors then it should just make them for you